servers/src/main/java/tachyon/master/next/filesystem/meta/InodeTree.java

```diff
@@ -220,7 +220,7 @@ public void deleteInode(Inode inode) throws FileDoesNotExistException {
 
     mInodes.remove(inode);
     mPinnedInodeFileIds.remove(inode.getId());
-    inode.reverseId();
+    inode.delete();
```

Comment:
That's a good point. In fact I think there might be a problem with the code - it didn't check whether the id is negative or not, and may result in reversing a deleted inode. So I think after the change it's cleaner .